storage/innobase/fil/fil0crypt.cc

```diff
@@ -1742,24 +2183,66 @@ static bool fil_crypt_start_rotate_space(const key_state_t *key_state,
   ut_ad(key_state->key_id == crypt_data->key_id);
 
   if (crypt_data->rotate_state.active_threads == 0) {
-    crypt_data->encrypting_with_key_version = key_state->key_version;
-    if (crypt_data->load_needed_keys_into_local_cache() == false) {
+    bool validation_tag_re_encryption_failure{false};
+    bool load_key_failure{false};
+    uint org_max_key_version = crypt_data->max_key_version;
+
+    if (crypt_data->load_keys_to_local_cache() == false) {
+      load_key_failure = true;
+    } else if (key_state->key_version != ENCRYPTION_KEY_VERSION_NOT_ENCRYPTED) {
+      if (crypt_data->re_encrypt_validation_tag(crypt_data->max_key_version + 1,
+                                                key_state->key_version) ==
+          false) {
+        validation_tag_re_encryption_failure = true;
```

Comment:
What do we do in this scenario? Ignore rotation for this space? Retry later?

Reply:
remove this space from rotation